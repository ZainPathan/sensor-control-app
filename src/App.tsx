import React, { useEffect, useState } from 'react';
import './App.style';
import { Grid } from '@material-ui/core';
import useAppStyles from './App.style';
import SensorCard from './components/SensorCard';
import HeaderSection from './components/HeaderSection';

const client = new WebSocket('ws://localhost:5000');

function App() {
    const classes = useAppStyles();
    const [isShowAllSensors, setShowAllSensors] = useState(true);

    const [temperatureSensor, setTemperatureSensor] = useState<any>({});
    const [pressureSensor, setPressureSensor] = useState<any>({});
    const [humiditySensor, setHumiditySensor] = useState<any>({});
    const [pm25Sensor, setPM25Sensor] = useState<any>({});
    const [pm10Sensor, setPM10Sensor] = useState<any>({});
    const [windSensor, setWindSensor] = useState<any>({});

    useEffect(() => {
        client.onopen = () => {
            console.log('WebSocket Client Connected');
            localStorage.removeItem('sensorData');
        };
        client.onmessage = (message) => {
            const currentSensorData: any = JSON.parse(message.data);

            switch(currentSensorData.name) {
                case 'Temperature':
                    setTemperatureSensor(currentSensorData);
                    break;
                case 'Pressure':
                    setPressureSensor(currentSensorData);
                    break;
                case 'Humidity':
                    setHumiditySensor(currentSensorData);
                    break;
                case 'PM2.5':
                    setPM25Sensor(currentSensorData);
                    break;
                case 'PM10':
                    setPM10Sensor(currentSensorData);
                    break;
                case 'Wind':
                    setWindSensor(currentSensorData);
                    break;
                default:
                    console.log('Ignored message: ', currentSensorData);
            }
        };
        return () => client.close();
    }, [])

    const handleSensorButtonClick = (currentSensorData: any) => {
        let cmdObj = {};
        if(currentSensorData.connected) {
            cmdObj = {
                "command": "disconnect",
                "id": currentSensorData.id
            };
        } else {
            cmdObj = {
                "command": "connect",
                "id": currentSensorData.id
            };
        }
        client.send(JSON.stringify(cmdObj));
    }

    return (
        <div className="App">
          <Grid container
                spacing={3}
          >
            <Grid item
                  xs={12}
            >
                <HeaderSection
                    handleSensorToggle={(isFlagSet: boolean) => {
                        setShowAllSensors(isFlagSet);
                    }}
                />
            </Grid>
            <Grid container
                  spacing={3}
                  className={classes.sensorCardsSection}
            >
                {/*Load Sensor cards here*/}
                {temperatureSensor && (!isShowAllSensors ? temperatureSensor.connected : true) && (<SensorCard
                    key={temperatureSensor.id}
                    sensorData={temperatureSensor}
                    handleSensorButtonClick={handleSensorButtonClick}
                />)}
                {pressureSensor && (!isShowAllSensors ? pressureSensor.connected : true) && (<SensorCard
                    key={pressureSensor.id}
                    sensorData={pressureSensor}
                    handleSensorButtonClick={handleSensorButtonClick}
                />)}
                {humiditySensor && (!isShowAllSensors ? humiditySensor.connected : true) && (<SensorCard
                    key={humiditySensor.id}
                    sensorData={humiditySensor}
                    handleSensorButtonClick={handleSensorButtonClick}
                />)}
                {pm25Sensor && (!isShowAllSensors ? pm25Sensor.connected : true) && (<SensorCard
                    key={pm25Sensor.id}
                    sensorData={pm25Sensor}
                    handleSensorButtonClick={handleSensorButtonClick}
                />)}
                {pm10Sensor && (!isShowAllSensors ? pm10Sensor.connected : true) && (<SensorCard
                    key={pm10Sensor.id}
                    sensorData={pm10Sensor}
                    handleSensorButtonClick={handleSensorButtonClick}
                />)}
                {windSensor && (!isShowAllSensors ? windSensor.connected : true) && (<SensorCard
                    key={windSensor.id}
                    sensorData={windSensor}
                    handleSensorButtonClick={handleSensorButtonClick}
                />)}
            </Grid>
          </Grid>
        </div>
    );
}

export default App;
