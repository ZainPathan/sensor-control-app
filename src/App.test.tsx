import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

describe('App component tests', () => {
    it('snapshot test', () => {
        const tree = renderer
            .create(
                <App />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});