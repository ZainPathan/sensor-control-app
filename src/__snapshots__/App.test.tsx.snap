// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component tests snapshot test 1`] = `
<div
  className="App"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        className="makeStyles-root-2"
      >
        <header
          className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-appBar-3 MuiPaper-elevation4"
        >
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
          >
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <h6
                className="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit"
              >
                Sensor Management
              </h6>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <label
                className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-align-items-xs-center"
              >
                <div
                  className="MuiGrid-root MuiGrid-item"
                >
                  Show Connected
                </div>
                <div
                  className="MuiGrid-root MuiGrid-item"
                >
                  <span
                    className="MuiSwitch-root"
                  >
                    <span
                      aria-disabled={false}
                      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-4 MuiSwitch-switchBase MuiSwitch-colorSecondary PrivateSwitchBase-checked-5 Mui-checked"
                      onBlur={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={null}
                    >
                      <span
                        className="MuiIconButton-label"
                      >
                        <input
                          checked={true}
                          className="PrivateSwitchBase-input-7 MuiSwitch-input"
                          onChange={[Function]}
                          type="checkbox"
                          value="checked"
                        />
                        <span
                          className="MuiSwitch-thumb"
                        />
                      </span>
                    </span>
                    <span
                      className="MuiSwitch-track"
                    />
                  </span>
                </div>
                <div
                  className="MuiGrid-root MuiGrid-item"
                >
                  Show All
                </div>
              </label>
            </div>
          </div>
        </header>
      </div>
    </div>
    <div
      className="MuiGrid-root makeStyles-sensorCardsSection-1 MuiGrid-container MuiGrid-spacing-xs-3"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 MuiGrid-grid-lg-4"
      >
        <div
          className="MuiPaper-root MuiCard-root MuiPaper-outlined MuiPaper-rounded"
        >
          <div
            className="MuiCardContent-root"
          >
            <h2
              className="MuiTypography-root MuiTypography-h5"
            />
            <h6
              className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
            >
              ID: 
            </h6>
            <h6
              className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Connection Status: 
              Disconnected
            </h6>
            <h6
              className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Value(in unit): 
              -
            </h6>
          </div>
          <div
            className="MuiCardActions-root MuiCardActions-spacing"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-8 MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Connect
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 MuiGrid-grid-lg-4"
      >
        <div
          className="MuiPaper-root MuiCard-root MuiPaper-outlined MuiPaper-rounded"
        >
          <div
            className="MuiCardContent-root"
          >
            <h2
              className="MuiTypography-root MuiTypography-h5"
            />
            <h6
              className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
            >
              ID: 
            </h6>
            <h6
              className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Connection Status: 
              Disconnected
            </h6>
            <h6
              className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Value(in unit): 
              -
            </h6>
          </div>
          <div
            className="MuiCardActions-root MuiCardActions-spacing"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-8 MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Connect
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 MuiGrid-grid-lg-4"
      >
        <div
          className="MuiPaper-root MuiCard-root MuiPaper-outlined MuiPaper-rounded"
        >
          <div
            className="MuiCardContent-root"
          >
            <h2
              className="MuiTypography-root MuiTypography-h5"
            />
            <h6
              className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
            >
              ID: 
            </h6>
            <h6
              className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Connection Status: 
              Disconnected
            </h6>
            <h6
              className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Value(in unit): 
              -
            </h6>
          </div>
          <div
            className="MuiCardActions-root MuiCardActions-spacing"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-8 MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Connect
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 MuiGrid-grid-lg-4"
      >
        <div
          className="MuiPaper-root MuiCard-root MuiPaper-outlined MuiPaper-rounded"
        >
          <div
            className="MuiCardContent-root"
          >
            <h2
              className="MuiTypography-root MuiTypography-h5"
            />
            <h6
              className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
            >
              ID: 
            </h6>
            <h6
              className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Connection Status: 
              Disconnected
            </h6>
            <h6
              className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Value(in unit): 
              -
            </h6>
          </div>
          <div
            className="MuiCardActions-root MuiCardActions-spacing"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-8 MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Connect
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 MuiGrid-grid-lg-4"
      >
        <div
          className="MuiPaper-root MuiCard-root MuiPaper-outlined MuiPaper-rounded"
        >
          <div
            className="MuiCardContent-root"
          >
            <h2
              className="MuiTypography-root MuiTypography-h5"
            />
            <h6
              className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
            >
              ID: 
            </h6>
            <h6
              className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Connection Status: 
              Disconnected
            </h6>
            <h6
              className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Value(in unit): 
              -
            </h6>
          </div>
          <div
            className="MuiCardActions-root MuiCardActions-spacing"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-8 MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Connect
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 MuiGrid-grid-lg-4"
      >
        <div
          className="MuiPaper-root MuiCard-root MuiPaper-outlined MuiPaper-rounded"
        >
          <div
            className="MuiCardContent-root"
          >
            <h2
              className="MuiTypography-root MuiTypography-h5"
            />
            <h6
              className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
            >
              ID: 
            </h6>
            <h6
              className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Connection Status: 
              Disconnected
            </h6>
            <h6
              className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Value(in unit): 
              -
            </h6>
          </div>
          <div
            className="MuiCardActions-root MuiCardActions-spacing"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-8 MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Connect
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
